sdk/monitor/azure-monitor-opentelemetry-exporter/tests/quickpulse/test_utils.py

```diff
@@ -425,3 +456,186 @@ def test_get_metrics_from_projections(self, projection_map_mock):
+
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._utils._validate_filter_predicate_and_comparand")
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._utils._validate_filter_field_name")
+    def test_validate_info_invalid_filters(self, field_mock, pred_mock):
+        valid_metric_info = mock.Mock(telemetry_type="Request", projection=None, filter_groups=[])
+        filter = mock.Mock()
+        filter_group = mock.Mock()
+        filter_group.filters = [filter]
+        valid_metric_info.filter_groups = [filter_group]
+        field_mock.return_value = True
+        pred_mock.return_value = False
+
+        self.assertFalse(_validate_derived_metric_info(valid_metric_info))
+
+    def test_valid_time_stamp_with_decimal_seconds(self):
+        time_stamp = "14.6:56:7.89"
+        expected_ms = 1234567890  # Expected milliseconds
+        self.assertEqual(_filter_time_stamp_to_ms(time_stamp), expected_ms)
+
+    def test_valid_time_stamp_with_whole_seconds(self):
+        time_stamp = "0.0:0:0.2"
+        expected_ms = 200  # Expected milliseconds
+        self.assertEqual(_filter_time_stamp_to_ms(time_stamp), expected_ms)
+
+    def test_valid_time_stamp_with_no_days(self):
+        time_stamp = "0.5:30:15.5"
+        expected_ms = 19815500  # 0 days, 5 hours, 30 minutes, 15.5 seconds
+        self.assertEqual(_filter_time_stamp_to_ms(time_stamp), expected_ms)
+
+    def test_valid_time_stamp_with_large_days(self):
+        time_stamp = "100.0:0:0"
+        expected_ms = 8640000000  # 100 days
+        self.assertEqual(_filter_time_stamp_to_ms(time_stamp), expected_ms)
+
+    def test_invalid_time_stamp(self):
+        time_stamp = "invalid_format"
+        self.assertIsNone(_filter_time_stamp_to_ms(time_stamp))
+
+    def test_time_stamp_missing_seconds(self):
+        time_stamp = "1.0:0:"
+        self.assertIsNone(_filter_time_stamp_to_ms(time_stamp))
+
+    def test_time_stamp_empty_string(self):
+        time_stamp = ""
+        self.assertIsNone(_filter_time_stamp_to_ms(time_stamp))
+
+    def test_no_filters(self):
+        data = mock.Mock()
+        result = _check_filters([], data)
+        self.assertTrue(result)
+
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._utils._check_any_field_filter")
+    def test_check_any_filters(self, check_mock):
+        filter_info = mock.Mock()
+        filter_info.field_name = "*"
+        filter_info.predicate = "Equal"
+        filter_info.comparand = "true"
+        data_mock = mock.Mock()
+        check_mock.return_value = True
+        result = _check_filters([filter_info], data_mock)
+        self.assertTrue(result)
+        check_mock.assert_called_once_with(filter_info, data_mock)
+
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._utils._check_custom_dim_field_filter")
+    def test_check_custom_dim_filters(self, check_mock):
+        filter_info = mock.Mock()
+        filter_info.field_name = "CustomDimensions."
```

Comment:
Wouldn't this just give you an empty custom dimension field name?
When a customer enters a custom dim in the UI, it usually appears like this in the response: CustomDimensions.<fieldEntered>

Reply:
Yes the purpose of this test was more to just check that `_check_custom_dim_field_filter` was called, but I changed the dimension to something meaningful regardless.